src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy

```diff
@@ -0,0 +1,155 @@
+package com.bugsnag.android.gradle
+
+import com.android.build.gradle.internal.core.Abi
+import com.android.build.gradle.internal.ndk.NdkHandler
+import org.apache.http.entity.mime.MultipartEntity
+import org.apache.http.entity.mime.content.FileBody
+import org.apache.http.entity.mime.content.StringBody
+import org.gradle.api.tasks.TaskAction
+/**
+    Task to upload shared object mapping files to Bugsnag.
+
+    Reads meta-data tags from the project's AndroidManifest.xml to extract a
+    build UUID (injected by BugsnagManifestTask) and a Bugsnag API Key:
+
+    https://developer.android.com/guide/topics/manifest/manifest-intro.html
+    https://developer.android.com/guide/topics/manifest/meta-data-element.html
+
+    This task must be called after shared object files are generated, so
+    it is usually safe to have this be the absolute last task executed during
+    a build.
+*/
+class BugsnagUploadNdkTask extends BugsnagUploadAbstractTask {
+    File intermediatePath
+    File symbolPath
+    String variantName
+    File projectDir
+    File rootDir
+    String toolchain
+
+    BugsnagUploadNdkTask() {
+        super()
+        this.description = "Generates and uploads the NDK mapping file(s) to Bugsnag"
+    }
+
+    @TaskAction
+    def upload() {
+
+        if (intermediatePath == null) {
+            return
+        }
+
+        // Look for the shared objects in likely folders
+        File binariesFile = null;
+        for (File intDir : intermediatePath.listFiles()) {
+            if (intDir.name.equals("cmake")) { // CMake
+                binariesFile = intDir;
+            } else if (intDir.name.equals("binaries")) { // Experimental
+                binariesFile = intDir;
+            } // TODO: where do the build-ndk files get output?
```

Comment:
Tested this, and added code to look for shared objects in <project>/obj/local which is where ndk-build seems to put them